### DISCUSSION TITLE
Customize autocomplete key

### DISCUSSION THREAD

**Discussion by moy2010** [nushell/nushell#6424] [Q&A]
In the [miscellaneous page](https://www.nushell.sh/cookbook/misc.html) it says that the keybindings for the autocomplete function can be changed in the `config.nu` file, but I can't find any up-to-date example. Also, since I want to change it to `Tab`, would it conflict with the keybinds for `completion_menu`?

```
    {
      name: completion_menu
      modifier: none
      keycode: tab
      mode: emacs # Options: emacs vi_normal vi_insert
      event: {
        until: [
          { send: menu name: completion_menu }
          { send: menunext }
        ]
      }
    }
```

Thanks in advance!

**Comment by reed9999** [comment]
As an aside, the link above doesn't presently work. I haven't spent enough time on the doc to know if it's been reorganized; in any case, it should probably redirect somewhere.

**Reply by amtoine** [reply]
i found [this page](https://www.nushell.sh/book/line_editor.html)

**Comment by lxl66566** [comment]
I use the following config:

```nu
$env.config.keybindings ++= [
    {
        name: take_history_hint
        modifier: none
        keycode: tab
        mode: [emacs, vi_normal]
        event: {
            until: [
                { send: historyhintwordcomplete }
            ]
        }
    }
];
```

but currently I can't find any doc of this config...
